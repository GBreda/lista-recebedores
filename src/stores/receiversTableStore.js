import { defineStore } from 'pinia'
import ReceiversService from '@/services/ReceiversService'

export const useReceiversTableStore = defineStore('receiversTableStore', {
  state: () => ({
    tableData: []
  }),
  actions: {
    async fetchTableData(payload) {
      try {
        const { data } = await ReceiversService.fetchReceivers(payload)

        this.tableData = data

        return Promise.resolve(data)
      } catch (error) {
        return Promise.reject(error)
      }
    }
  }
})
