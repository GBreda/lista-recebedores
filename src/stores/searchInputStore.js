import { defineStore } from 'pinia'

export const useSearchInputStore = defineStore('searchInputStore', {
  state: () => ({
    input: ''
  }),
  actions: {
    setSearchInputValue(value) {
      this.input = value
    }
  }
})
