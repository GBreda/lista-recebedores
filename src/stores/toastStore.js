import { defineStore } from 'pinia'

export const useToastStore = defineStore('toastStore', {
  state: () => ({
    toastInfo: {
      showToast: false,
      message: '',
      kind: ''
    }
  }),
  actions: {
    setToastInfo(value) {
      this.toastInfo = value
    }
  }
})
