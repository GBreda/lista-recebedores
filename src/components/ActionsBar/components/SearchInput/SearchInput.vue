<template>
  <div class="search">
    <input
      :value="inputValue"
      @input="updateInput"
      class="search__input"
      type="text"
      placeholder="Nome, CPF, agÃªncia ou conta"
      data-test-id="search-input"
    />
    <img src="@/assets/images/search-icon.png" class="search__icon" />
  </div>
</template>

<script setup>
import { ref } from 'vue'
import { useSearchInputStore } from '@/stores/searchInputStore'

const searchInputStore = useSearchInputStore()

const inputValue = ref('')

const updateInput = (event) => {
  searchInputStore.setSearchInputValue(event.target.value)
}
</script>

<style lang="scss" scoped>
.search {
  position: relative;
  display: flex;
  align-items: center;

  &__input {
    width: 25rem;
  }

  &__icon {
    position: absolute;
    width: 1.6rem;
    right: 1rem;
  }
}
</style>
