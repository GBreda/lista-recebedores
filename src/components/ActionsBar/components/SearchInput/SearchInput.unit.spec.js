import { describe, it, expect, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import { createTestingPinia } from '@pinia/testing'
import { useSearchInputStore } from '@/stores/searchInputStore'
import SearchInput from './SearchInput.vue'

const mountSearchInput = () => {
  return mount(SearchInput, {
    global: {
      plugins: [
        createTestingPinia({
          createSpy: vi.fn
        })
      ]
    }
  })
}

describe('SearchInput', () => {
  let wrapper

  it('should mount the component', () => {
    wrapper = mountSearchInput()

    expect(wrapper.vm).toBeDefined()
  })

  it('should match the snapshot', () => {
    wrapper = mountSearchInput()

    expect(wrapper.element).toMatchSnapshot()
  })

  it('should emit input event on search input', async () => {
    wrapper = mountSearchInput()

    const store = useSearchInputStore()

    const input = wrapper.find("[data-test-id='search-input']")

    input.setValue('conta x')

    expect(store.setSearchInputValue).toHaveBeenCalled()
    expect(store.setSearchInputValue).toHaveBeenLastCalledWith('conta x')
  })
})
