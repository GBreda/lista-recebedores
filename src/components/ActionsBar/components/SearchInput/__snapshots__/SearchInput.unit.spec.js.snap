// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchInput > should match the snapshot 1`] = `
<div
  class="search"
  data-v-4a4b4034=""
>
  <input
    class="search__input"
    data-test-id="search-input"
    data-v-4a4b4034=""
    placeholder="Nome, CPF, agÃªncia ou conta"
    type="text"
  />
  <img
    class="search__icon"
    data-v-4a4b4034=""
    src="/src/assets/images/search-icon.png"
  />
</div>
`;
